### PR TITLE
test: mark skipped tests as SKIPPED

### LINKED ISSUE #21828: mark skipped tests as SKIPPED, not SUCCESS
Inspired by
https://github.com/systemd/systemd/pull/21797#issuecomment-996077793

I think we should mark skipped tests accordingly in the summary. Otherwise it gives the wrong impression that tests have actually successfully been run.
An example for this misconception can be found at
https://github.com/systemd/systemd/pull/21253#issuecomment-979460864

Not blaming @poettering here, but our CI infra :-)

### PR DESCRIPTION
**Pull request by yuwata**
Closes #21828.

### CODE REVIEW

File: test/run-integration-tests.sh [hunk 1]
```diff
@@ -47,7 +47,9 @@ declare -A results
 declare -A times
 
 COUNT=0
-FAILURES=0
+COUNT_SUCCESS=0
+COUNT_SKIPPED=0
+COUNT_FAILURES=0
```
Comment (medhefgo) [inline test/run-integration-tests.sh]: Imho, this should be called `COUNT_FAILED`. It fits with skipped and rolls better off the tongue.

Comment (yuwata) [reply test/run-integration-tests.sh]: Thanks. Addressed.

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
rpm-build* is failing but will be fixed by #21834.

**Comment by keszybz** [comment]
This doesn't seem to work…
```
TEST-55-OOMD was skipped:
no cgroup2
no oomd
Spawning getter /bin/journalctl -o export -D /var/tmp/systemd-test.t8lz1B/root/var/log/journal/4676accff3f64b2aae070cb2c1ffc699...
Finishing after writing 1696 entries
---- /var/tmp/systemd-test.t8lz1B/root/var/log/journal/4676accff3f64b2aae070cb2c1ffc699 ----
[  825.664658] systemd-tmpfiles[24]: Cannot set file attributes for '/var/log/journal', maybe due to incompatibility in specified attributes, previous=0x00080000, current=0x00080000, expected=0x00880000, ignoring.
[  825.664821] systemd-tmpfiles[24]: Cannot set file attributes for '/var/log/journal/4676accff3f64b2aae070cb2c1ffc699', maybe due to incompatibility in specified attributes, previous=0x00080000, current=0x00080000, expected=0x00880000, ignoring.
-rw-r-----+ 1 root root 8388608 Dec 20 13:13 /tmp/autopkgtest.jW3CAR/upstream-artifacts/TEST-55-OOMD.journal
nspawn-root test failed
make: *** [Makefile:4: run] Error 1
make: Leaving directory '/tmp/autopkgtest.jW3CAR/build.iYi/systemd/test/TEST-55-OOMD'
[13:13:21] --x-- Result of TEST-55-OOMD: 2 --x--
```

**Comment by mrc0mmand** [comment]
> Or maybe the ubuntu tests are using a different runner?

IIRC they shouldn't https://salsa.debian.org/systemd-team/systemd/-/blob/upstream-ci/debian/tests/upstream

**Comment by yuwata** [comment]
I hope the issue is now fixed. Setting the green label again.

**Comment by bluca** [comment]
Still unhappy:

```
TEST-55-OOMD was skipped:
no cgroup2
no oomd
Spawning getter /bin/journalctl -o export -D /var/tmp/systemd-test.0ZsM8K/root/var/log/journal/10d70e776f7f437ca1ba928a3fd85e35...
Finishing after writing 1710 entries
---- /var/tmp/systemd-test.0ZsM8K/root/var/log/journal/10d70e776f7f437ca1ba928a3fd85e35 ----
[  677.214423] systemd-tmpfiles[24]: Cannot set file attributes for '/var/log/journal', maybe due to incompatibility in specified attributes, previous=0x00080000, current=0x00080000, expected=0x00880000, ignoring.
[  677.214633] systemd-tmpfiles[24]: Cannot set file attributes for '/var/log/journal/10d70e776f7f437ca1ba928a3fd85e35', maybe due to incompatibility in specified attributes, previous=0x00080000, current=0x00080000, expected=0x00880000, ignoring.
-rw-r-----+ 1 root root 8388608 Dec 20 14:38 /tmp/autopkgtest.sPQzEK/upstream-artifacts/TEST-55-OOMD.journal
make: Leaving directory '/tmp/autopkgtest.sPQzEK/build.Kmb/systemd/test/TEST-55-OOMD'
make: *** [Makefile:4: run] Error 77
[14:38:06] --x-- Result of TEST-55-OOMD: 2 --x--
```

```
TEST-55-OOMD:                       FAIL     (  3 s)
```

**Comment by yuwata** [comment]
oh, `make` does not propagate the error code? hmm...

**Comment by mbiebl** [comment]
> oh, `make` does not propagate the error code? hmm...

could we check for the presence of /skipped ?

**Pull request closed without merging**